### ISSUE TITLE
Inclusion of indirect `"LinkingTo"` dependencies in lockfile

### ISSUE THREAD

**Issue by nbenn** [r-lib/pak#485]
Consider the following setup: Root package {tidyverse} `Imports` {readxl} `LinkingTo` {cpp11}.

In a lockfile generated as

```r
lockfile_create(
  "tidyverse",
  dependencies = list(
    direct = "Imports",
    indirect = "LinkingTo"
  )
)
```

Am I wrong to expect {cpp11} in there, irrespectively of whether my OS/repos setting supports binary installs?

**Comment by nbenn** [comment]
Thanks for getting back to me. Is there a way to disable this, i.e. to force inclusion of such dependencies?

It seems to me that currently there is no consensus on whether or not such dependencies are actually optional or not. For example, when using pak in a CI setup that goes on to deploy to connect, absence of such dependencies seems to be a problem. Deployment fails with

```
...
* Content depends on package "cpp11" but it is not installed. Please
  resolve before continuing.
...
```

Happy to provide more details or try to come up with a reproducible example for the issue I'm describing if you feel this is worth looking into.

**Comment by gaborcsardi** [comment]
If you want to use the lockfile on another platform, then create it for that platform. E.g. for linux set the platform to `source` before creating the lockfile:

```r
options(pkg.platforms = "source")
```

**Comment by gaborcsardi** [comment]
Can you tell me more how you use packrat to install the packages in the pak lockfile?

**Comment by nbenn** [comment]
I don't use packrat to install anything.

I use pak to install packages on my CI system which then runs checks and if successful, initiates deployment to Connect. For that I need a `manifest.json`, which [per docs](https://docs.posit.co/connect/user/publishing-cli-manifest/) can be created using `rsconnect::writeManifest()`. This internally creates a [packrat "snapshot"](https://github.com/rstudio/rsconnect/blob/72bcf625ca0805629838a41755b7fe5bfd8d12e7/R/bundlePackage.R#L261-L267).

The deployment however fails with messages like

```
* Content depends on package "cpp11" but it is not installed. Please
  resolve before continuing.
```

I haven't pinpointed where exactly they are coming from, but what I believe is happening is that packrat considers indirect "LinkingTo" dependencies as "hard" and refuses to go on because pak does not (in this specific set up) and they consequently were not installed.

One way to fix this would be to have pak install indirect "LinkingTo" dependencies even if they are not strictly required, ideally using binary releases (if possible) instead of sources.

Does that make sense? I'm happy to look into this further if you're interested. But as said earlier, I'd find it perfectly understandable if you tended towards something along the lines of "this is more the problem of packrat/Connect/etc. than that of pak".

**Comment by gaborcsardi** [comment]
That makes sense. It is unlikely that this would be fixed in packrat I think, but we can try to support it in pak.

**Comment by nbenn** [comment]
Yeah, sounds great to me! Thanks.

**Comment by gaborcsardi** [comment]
Closed by #316. Will be in tomorrow's nightly pak build.
